src/basilisp/lang/reader.py

```diff
@@ -1696,9 +1707,14 @@ def read(  # pylint: disable=too-many-arguments
     features: Optional[IPersistentSet[kw.Keyword]] = None,
     process_reader_cond: bool = True,
     default_data_reader_fn: Optional[DefaultDataReaderFn] = None,
+    line_num: Optional[int] = None,
+    column_num: Optional[int] = None,
 ) -> Iterable[RawReaderForm]:
     """Read the contents of a stream as a Lisp expression.
 
+    The optional `line_num` and `column_num` specify where the
+    `stream` starts in the broader context, if not from the start.
```

Comment:
```suggestion
    The optional `line_num` and `column_num` specify where the `stream`
    location metadata starts in the broader context, if not from the start.
```

Same suggestion for `read_str` below, if you don't mind.

Reply:
Updated.